contracts/MechMarketplace.sol

```diff
@@ -206,7 +206,7 @@ contract MechMarketplace is IErrorsMarketplace {
     function _deliverMarketplaceWithSignatures(
         address requester,
         bytes32 paymentType,
-        DeliverWithSignature[] calldata deliverWithSignatures,
+        DeliverWithSignature[] memory deliverWithSignatures,
```

Comment:
This change is needed for addressing the "stack too deep" error otherwise. I don't see an issue here as the function is internal.